scripts/Clone-Repos.ps1

```diff
@@ -22,6 +22,11 @@ $repos = ($reposResponse.Content | ConvertFrom-Json).value
 foreach ($repo in $repos) {
 	$repoPath = $repo."$repoPathProperty"
 	$repoLocation = "$basePath\$($repo.name)"
+
+	if ($cloneWithHttps) {
+		$repoPath = $repoPath.Replace("https://$($project)@","https://$($project):$($pat)@")
```

Comment:
Technically the part of the URL represented by `$project` is the user name. It defaults to the project name but can be anything according to [this article](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops#use-your-personal-access-token)

Reply:
What would you propose to change it to? "anything" ? :P

Reply:
If you are afraid that the API will not return the default username in form of project I can complicate the replacement logic - is that the point?

Reply:
Well I don't think the API would return anything other than the defaults. I would either leave it as is or use a capturing group named "user" in a regular expression.

Reply:
Check it I used a wildcard to capture anything between `https://` and the `@` sign. So even if it would be some other credentials, it will replace the whole thing.